src/ngraph/runtime/cpu/op/loop_kernel.hpp

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "ngraph/op/util/requires_tensor_view_args.hpp"
+#include "ngraph/util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace cpu
+        {
+            namespace op
+            {
```

Comment:
This needs a description of what the op does. I think just computes outputs, the same as would have happened without it, only it does the part from args to outputs more efficiently.
Couldn't node_list be computed from outputs and args?

Reply:
Will add the description. 

> Couldn't node_list be computed from outputs and args?

Yes, but `node_list` will be anyways constructed in a topologically sorted order by a fusion pass. It seems more practical to simply "cache" it inside an op. This way we don't need more logic to do the traversals from the outputs to args and also we save ourselves a computation. I'd suggest we will leave it this way until I add the fusion pass.